velox/exec/tests/RowNumberTest.cpp

```diff
@@ -265,4 +265,51 @@ TEST_F(RowNumberTest, maxSpillBytes) {
   }
 }
 
+TEST_F(RowNumberTest, memoryUsage) {
+  const auto rowType =
+      ROW({"c0", "c1", "c2"}, {INTEGER(), INTEGER(), VARCHAR()});
+  const auto vectors = createVectors(rowType, 1024, 100 << 20);
+  core::PlanNodeId rowNumberId;
+  auto plan = PlanBuilder()
+                  .values(vectors)
+                  .rowNumber({"c0", "c2"})
+                  .capturePlanNodeId(rowNumberId)
+                  .project({"c0", "c1"})
+                  .planNode();
+
+  int64_t peakBytesWithSpilling = 0;
+  int64_t peakBytesWithOutSpilling = 0;
+
+  for (const auto& spillEnable : {false, true}) {
+    auto queryCtx = std::make_shared<core::QueryCtx>(executor_.get());
+    auto spillDirectory = exec::test::TempDirectoryPath::create();
+    const std::string spillEnableConfig = std::to_string(spillEnable);
+
+    std::shared_ptr<Task> task;
+    AssertQueryBuilder(plan)
+        .spillDirectory(spillDirectory->path)
+        .queryCtx(queryCtx)
+        .config(core::QueryConfig::kSpillEnabled, spillEnableConfig)
```

Comment:
It would be nice to add overrides of .config(bool|int64_t) to make it easier to write tests

```
.config(core::QueryConfig::kSpillEnabled, spillEnable)
.config(core::QueryConfig::kTestingSpillPct, 100)
```

Reply:
That would be great. Thank you.